### PR TITLE
Add a styled Modal

### PR DESCRIPTION
**Pull request by meslater1030**
This PR:

* Adds a basic modal to Varnish. This is an Ant modal with Varnish buttons and a couple of other adjustments.
* Adds a demo of that modal
* This PR does not include any additional styling for `Modal.info`, `Modal.error` etc.

GIF:
Modal Demo

Still:
Screen Shot 2019-08-09 at 10 02 25 AM

### CODE REVIEW

File: ui/src/demo/pages/Design/Design.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
     Headers,
     Icons,
     Links,
+    Modals,
```
Comment (jonborchardt) [inline ui/src/demo/pages/Design/Design.tsx]: please rename modal (the plural ones are incorrect)

File: ui/src/demo/pages/index.ts [hunk 2]
```diff
@@ -11,3 +11,4 @@ export * from './Design/Palette';
 export * from './Design/Spacing';
 export * from './Design/Typography';
 export * from './Design/Tables';
+export * from './Design/Modals';
```
Comment (jonborchardt) [inline ui/src/demo/pages/index.ts]: please rename Modal

File: ui/src/lib/components/index.ts [hunk 3]
```diff
@@ -9,3 +9,4 @@ export * from './MaxWidthCenteredContent';
 export * from './typography';
 export * from './icon';
 export * from './Table';
+export * from './Modal';
```
Comment (jonborchardt) [inline ui/src/lib/components/index.ts]: please add a new line at end of file

File: ui/src/demo/pages/Design/Modals.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { BodySmall, ExternalLink } from '../../../lib/components';
```
Comment (jonborchardt) [inline ui/src/demo/pages/Design/Modals.tsx]: new line after external references (after react router)

File: ui/src/lib/components/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import styled from 'styled-components';
+import { Modal as AntModal } from 'antd';
+import { ModalProps } from 'antd/lib/modal';
+import { Button, ButtonVariant } from './button';
+
+import React from 'react';
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: put react at top of file

File: ui/src/lib/components/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import styled from 'styled-components';
+import { Modal as AntModal } from 'antd';
+import { ModalProps } from 'antd/lib/modal';
+import { Button, ButtonVariant } from './button';
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: new line after external references

File: ui/src/lib/components/Modal.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    );
+  }
+}
+
+const StyledAntModal = styled(AntModal)`
+  .ant-modal-close {
+    display: none;
+  }
+  .ant-modal-header {
+    background-color: ${props => props.theme.palette.background.light.hex};
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: remove .hex

File: ui/src/lib/components/Modal.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    border-bottom: 1px solid ${props => props.theme.palette.border.main.hex};
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: remove .hex

File: ui/src/lib/components/Modal.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    padding: 16px;
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: padding should be theme.spacing.md

File: ui/src/lib/components/Modal.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  .ant-modal-body {
+    padding: 16px;
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: padding should be theme.spacing.md

File: ui/src/lib/components/Modal.tsx [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  .ant-modal-footer {
+    border-top: none;
+    text-align: left;
+    padding: 16px;
```
Comment (jonborchardt) [inline ui/src/lib/components/Modal.tsx]: padding should be theme.spacing.md

**Comment by jonborchardt** [comment]
please also send @Pelagic-sky a request for visual feedback?

**Comment by Pelagic-sky** [comment]
LGTM

**Pull request merged**